File: JavaScript/107-Binary Tree Level Order Traversal II.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrderBottom = function(root) {
    let res = new Array();
    
    let dfs = (node, level) =>{
        if (!node) return;
        if (res.length === level) res.push(new Array());
        res[level].push(node.val);
        
        dfs(node.left, level+1);
        dfs(node.right, level+1);
    }
    dfs(root, 0);
    return res.reverse();
};

File: 107-Binary Tree Level Order Traversal II.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def levelOrderBottom(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
# preorder transversal
        res = []
        
        def dfs(node, level):
            if not node:
                return
            if len(res) == level:
                res.append([])
            res[level].append(node.val)
            dfs(node.left, level+1)
            dfs(node.right, level+1)
            
        dfs(root, 0)
        return res[::-1]